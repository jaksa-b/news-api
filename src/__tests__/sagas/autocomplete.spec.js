import { expectSaga } from "redux-saga-test-plan";

import watchAutocompletes from "../../redux/sagas/autocompletes";

describe("Autocompletes saga", () => {
  it("should have the expected watchers", (done) => {
    expectSaga(watchAutocompletes)
      .run({ silenceTimeout: true })
      .then((saga) => {
        expect(saga).toMatchSnapshot();
        done();
      });
  });
});
