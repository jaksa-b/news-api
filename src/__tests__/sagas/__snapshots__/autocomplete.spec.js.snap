// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Autocompletes saga should have the expected watchers 1`] = `
Object {
  "fork": Array [
    Object {
      "@@redux-saga/IO": true,
      "combinator": false,
      "payload": Object {
        "args": Array [
          "FETCH_AUTOCOMPLETES_REQUEST",
          "@@redux-saga-test-plan/json/function/fetchAutocompletes",
        ],
        "context": null,
        "fn": "@@redux-saga-test-plan/json/function/takeLatest",
      },
      "type": "FORK",
    },
  ],
  "take": Array [
    Object {
      "@@redux-saga/IO": true,
      "combinator": false,
      "payload": Object {
        "pattern": "FETCH_AUTOCOMPLETES_REQUEST",
      },
      "type": "TAKE",
    },
  ],
}
`;
